community-tutorials/setup-wireguard/01-en.md

```diff
@@ -0,0 +1,150 @@
+> WireGuard® is an extremely simple yet fast and modern VPN that utilizes state-of-the-art cryptography. 
+
+In this tutorial we focus on a simple setup with one server and one client peer. 
+
+# Requirements
+- A server (root or VPS)
+- Ubuntu as an operating system (debian would also work)
+
+# Step 1 - Install WireGuard®
+
+First we need to install WireGuard® on our new server. For Ubuntu this is quite easy:
+
+```bash
+sudo apt install wireguard
+```
+
+# Step 2 - Generate keys
+
+WireGuard® works with private and public keys. For every client, we need one private and one public key. WireGuard® treats the server itself as a client as well, so we also need private/public keys for the server.
+
+To generate these keys, we can use the following command:
+```bash
+wg genkey | tee server-private.key | wg pubkey > server-public.key
+wg genkey | tee client-private.key | wg pubkey > client-public.key
+```
+
+We now have a private/public key for the server and a private/public key for one client. 
+
+# Step 3 - Configuration of the WireGuard® server
+
+To configure the WireGuard® server, we need to edit the following file:
+```bash
+sudo nano /etc/wireguard/wg0.conf
+```
+
+The content of this file should look like this:
+```conf
+[Interface]
+Address = 10.0.0.1/24
+PrivateKey = <server-private.key>
+ListenPort = 51820
+SaveConfig = true
+PostUp = iptables -A FORWARD -i wg0 -j ACCEPT; iptables -t nat -A POSTROUTING -o eth0 -j MASQUERADE; ip6tables -A FORWARD -i wg0 -j ACCEPT; ip6tables -t nat -A POSTROUTING -o eth0 -j MASQUERADE
+PostDown = iptables -D FORWARD -i wg0 -j ACCEPT; iptables -t nat -D POSTROUTING -o eth0 -j MASQUERADE; ip6tables -D FORWARD -i wg0 -j ACCEPT; ip6tables -t nat -D POSTROUTING -o eth0 -j MASQUERADE
+
+[Peer]
+PublicKey = <client-public.key>
+AllowedIPs = 10.0.0.2/32
+```
+
+- `Address` will set the internal IP address range of the WireGuard® network
+- `PrivateKey` should be the `server-private.key` we generated earlier
+- `ListenPort` defines the port of the WireGuard® server
+- `SaveConfig` defines, if the current config should be saved, when the service is stopped
+- `PostUp` ensures, that the firewall will not block the connection, when the WireGuard® server is started
+- `PostDown` removes the firewall rules, once the WireGuard® service is stopped
+- For every client in the network, one `[Peer]` element should be created within this file (therefore only one in our example)
+- `PublicKey` should contain the `client-public.key`, we generated earlier
+- `AllowedIPs` defines all IPs that this peer is allowed to have when connected to the network
+
+## Step 3.1 - Set IP forwarding
+
+We also need to set IP V4 forwarding, in order to make WireGuard® work on our Ubuntu Netcup server:
+```diff
+sudo nano /etc/sysctl.conf
+- #net.ipv4.ip_forward=1
++ net.ipv4.ip_forward=1
+```
+
+# Step 4 - Configure client
+
+Next to the sever, we also need to configure our client.
```

Comment:
Where do I have to put this file?

Reply:
I added two more steps that explain, how to configure the client via GUI or config file